tests/utils/mocks/mock_objects.py

```diff
@@ -0,0 +1,233 @@
+        return population
+
+    @staticmethod
+    def create_clustered_population(
+        size: int, dim: int, num_clusters: int = 3
+    ) -> List[GaIndividual]:
+        """
+        Create a population with clustered individuals.
+
+        :param size: Population size
+        :param dim: Chromosome dimension
+        :param num_clusters: Number of clusters
+        :return: List of individuals
+        """
+        population = []
+        individuals_per_cluster = size // num_clusters
+
+        for cluster in range(num_clusters):
+            # Create cluster center
+            center = np.random.rand(dim)
+
+            for _ in range(individuals_per_cluster):
+                # Add noise around cluster center
+                noise = np.random.normal(0, 0.1, dim)
+                chromosome = center + noise
+                # Clamp to [0, 1]
+                chromosome = np.clip(chromosome, 0.0, 1.0)
+                individual = GaIndividual(real_chrom=chromosome)
+                population.append(individual)
+
+        # Add remaining individuals if size is not divisible by num_clusters
+        remaining = size - len(population)
+        for _ in range(remaining):
+            chromosome = np.random.rand(dim)
+            individual = GaIndividual(real_chrom=chromosome)
+            population.append(individual)
+
+        return population
+
+    @staticmethod
+    def create_known_optima_data(dim: int) -> dict:
+        """
+        Create individuals at known optima for benchmark functions.
+
+        :param dim: Problem dimension
+        :return: Dictionary with optima for different functions
+        """
+        return {
+            "sphere_optimum": GaIndividual(real_chrom=np.zeros(dim)),
+            "rastrigin_optimum": GaIndividual(real_chrom=np.zeros(dim)),
+            "rosenbrock_optimum": GaIndividual(real_chrom=np.ones(dim)),
+            "random_point": GaIndividual(real_chrom=np.random.rand(dim)),
+            "boundary_point_lower": GaIndividual(real_chrom=np.full(dim, -5.0)),
+            "boundary_point_upper": GaIndividual(real_chrom=np.full(dim, 5.0)),
+        }
+
+
+class TestScenarios:
+    """Predefined test scenarios for common testing patterns."""
+
+    @staticmethod
+    def optimization_convergence_scenario(
+        evaluator, initial_population: List[GaIndividual]
+    ) -> dict:
+        """
+        Create a scenario for testing optimization convergence.
+
+        :param evaluator: Function evaluator
+        :param initial_population: Starting population
+        :return: Dictionary with scenario data
+        """
+        # Evaluate initial population
+        initial_fitness = []
+        for individual in initial_population:
+            args = GaEvaluatorArgs(individual)
+            fitness = evaluator.evaluate(args)
+            initial_fitness.append(fitness)
+
+        best_initial_idx = np.argmin(initial_fitness)
+        worst_initial_idx = np.argmax(initial_fitness)
```

Comment:
Here is an assumption that lower fitness score means better, but if `evaluator` returns `GaExtremum.MAXIMUM`, then it will be the opposite. Of course, currently all benchmark functions (Rastrigin, Rosenbrock and Sphere) return `GaExtremum.MINIMUM`, but it will not be guaranteed in the future.